parts/kubernetesagentcustomdata.yml

```diff
@@ -125,6 +125,11 @@ write_files:
     KUBELET_FIX_43704_3=""
   {{end}}
 {{end}}
+{{if UseCloudControllerManager }}
+    CLOUD_PROVIDER=external
+{{else}}
+    CLOUD_PROVIDER=azure
+{{end}}
 
 - path: "/etc/systemd/system/kubelet.service"
```

Comment:
Is there no change needed for `kubelet.service` to include the `--provider-id` flag?

Reply:
provider-id can be detected now:
https://github.com/kubernetes/kubernetes/pull/50730
https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/azure/azure_instances.go#L103